### PR TITLE
feat: fastify support

### LINKED ISSUE #2240: [✨] Integration with Fastify
### Is your feature request related to a problem?

We are support already netlify, vercel and cloudflare! we want more deployment integrations. In order to add a new one folder needs to be created here, with the necesary configuration for Fastify:
https://github.com/BuilderIO/qwik/tree/main/starters/adaptors

### Describe the solution you'd like

Nop

### Describe alternatives you've considered

Nop

### Additional context

Join the discord and text in the #contributions channel! the core team will be happy to guide you make the PR!

### PR DESCRIPTION
**Pull request by ilteoood**
# What is it?

- [X] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests

# Description

closes #2240

- feat: support for `fastify` server
 
# Use cases and why

- 1. Use `fastify` as server
- 2. [Increase performances](https://www.fastify.io/benchmarks/)

# Checklist:

- [X] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by zanettin** [comment]
Thank you very much @ilteoood 🙏 waiting for a approval of the core team 🎉

**Review by ilteoood — COMMENTED** [review]
I'll start to re-align the PR

**Comment by manucorporat** [comment]
Hi! thanks for the PR! unfortunately, we cant merge this because of the breaking change, specially since it could be easily prevented with some deprecated APIs.

We should add back to the exports in the package.json the exports for the express, so previous code still works. Also the renamed methods should stay with a deprecated annotation

File: scripts/qwik-city.ts
Comment (manucorporat) [inline scripts/qwik-city.ts:70]: we need to have two modules pointing to the same new file, to prevent breaking changes

File: scripts/qwik-city.ts
Comment (manucorporat) [inline scripts/qwik-city.ts:112]: same here, we cant remove this

File: packages/qwik-city/adapters/node-server/vite/index.ts
Comment (manucorporat) [inline packages/qwik-city/adapters/node-server/vite/index.ts:38]: we cant't remove this lines

**Review by manucorporat — CHANGES_REQUESTED** [review]

**Comment by ilteoood** [comment]
Hi @manucorporat, thank you so much for your review.

To be honest, I don't agree with what you said, because the qwik API is not yet stable, since the project is still in version 0, and this is the right time to introduce breaking changes.

I personally don't like to have deprecated code while the API is unstable, because it will be much harder in the future to remove all the deprecated code before the stable release.

By the way, I think I'll recreate the PR from scratch in order to have the minimum amount of conflict. Please be patient 🙏

**Comment by ilteoood** [comment]
Ping @manucorporat

File: scripts/qwik-city.ts
Comment (manucorporat) [inline scripts/qwik-city.ts:72]: this files are not longer generated

File: scripts/qwik-city.ts
Comment (manucorporat) [inline scripts/qwik-city.ts:409]: shouldn't we add a new `buildAdapterNodeServer` instead o replacing express? otherwise we will not generate the files, that the package.json is pointing to

**Comment by manucorporat** [comment]
Dealing with broken code when it can be easily avoided is day saver for us. I agree we can break now, but if we can easily avoid, i would prefer. Let's give developers a way to know what deprecated so, they know how to update, vs suddenly not working.

**Comment by manucorporat** [comment]
This is an amazing PR btw! let's ship it!

**Review by manucorporat — APPROVED** [review]

**Comment by manucorporat** [comment]
In other to show fastify in the next release we need to add a e2e test here:
https://github.com/BuilderIO/qwik-city-e2e

**Comment by manucorporat** [comment]
basically, it's deploying the exact app in fastify and run all the tests we already have!

**Comment by Dindaleon** [comment]
Are we using fastify as default now or we need to activate it somewhere?

**Pull request merged**
